central/resourcecollection/service/service_impl.go

```diff
@@ -203,7 +204,8 @@ func (s *serviceImpl) UpdateCollection(ctx context.Context, request *v1.UpdateCo
 }
 
 func collectionRequestToCollection(ctx context.Context, request collectionRequest, id string) (*storage.ResourceCollection, error) {
-	if request.GetName() == "" {
+	collectionName := strings.TrimSpace(request.GetName())
```

Comment:
Does this strip the leading/trailing whitespace before attempting to save the collection? Would it be more predictable for API users if this rejected the request with an error instead?

It is probably unlikely, but it seems possible this could break an automation if the name the user submits via script is different from the name that is stored in the database.

Reply:
The function is used by all endpoints as a generalized operation to convert a request object to a storage object. So it would trim leading/trailing whitespace before every operation (get, delete, update, create, dryrun)

Reply:
I wouldn't worry about failures due to misconfiguration. The result of APIs is reproducible and not a one-off.